# Resources for Others
This repository contains a smattering of tutorials and example documents that I've created to make others' lives a bit easier than mine was in trying to learn LaTeX and other things.

## LaTeX Templates
1. **Template for Writing** - A general template for writing a manuscript, etc. You can either download the .zip file included [here](https://github.com/jessicarick/resources/blob/master/Template%20for%20Writing%20-%20Public.zip), or clone the Overleaf project at: https://www.overleaf.com/read/fqsfnqxsqbxf.
2. **Preprint Template** - A modified version of my "Template for Writing," designed for manuscript preprints. You can clone the Overleaf project at: https://www.overleaf.com/read/bpsxspkmhvhf.
3. **Preprint Template II** - A different version of a preprint/manuscript template, with sans serif font and a separate title page. You can clone the Overleaf project or download the source files at: https://www.overleaf.com/read/ghcstpqfynqm -- or you can download the zipped source files [here](https://github.com/jessicarick/resources/blob/master/Preprint%20Template%20-%20Public.zip).
4. **Proposal Template** - A generic template for writing a thesis or dissertation proposal. You can either download the .zip file [here](https://github.com/jessicarick/resources/blob/master/Proposal%20Template%20-%20Public.zip), or clone the Overleaf project at: https://www.overleaf.com/read/btddnrrztkgz.
5. **Research Log Template** - A template for keeping track of bioinformatics work (or any kind of research). Keeping a “lab notebook” for computer-based analyses is crucial for open and reproducible science, and has made a HUGE difference for me in streamlining my analyses (no matter how many times I have to re-do them). Different people have different things that work for them, but this is what has worked for me for integrating code, figures, and commentary across different types of analysis (i.e. some cluster computing, some R analysis, some GUI program work, etc). You can either download the .zip file [here](https://github.com/jessicarick/resources/blob/master/Research%20Log%20Template%20-%20Public.zip), or clone the Overleaf project at: https://www.overleaf.com/read/wkxxnwvfbdxn.
6. **Academic CV Template** - This is a general template for a CV, geared toward the style of CV used in academia in the (biological) sciences. Everyone has different opinions on how CVs should look, but this is what I've come up with for myself (you can see what my CV using this template looks like here: [http://www.jessicarick.com/lab-members](https://jessicarick.com/lab-members)). You can either download the source files [here](https://github.com/jessicarick/resources/blob/master/CV%20Template%20-%20Public.zip), or use the Overleaf template that I created, found here: https://www.overleaf.com/latex/examples/academic-cv-template/hvjpfjnyggbf.

## Tutorials
1. **Integrating R Analyses with Overleaf** - I wrote up this tutorial to explain how to incorporate R Markdown documents into Overleaf (V2) documents, as it pretty much blew my mind when I learned that the two could talk to each other. Just think, you can run your analyses in your R Markdown file, create figures, and they’ll update in your manuscript on Overleaf! What more could you want in life?!? PDF document with tutorial (R_Overleaf_Integration.pdf) and associated trimtex.sh script, as well as latex-rmd.sty package can be downloaded [here](https://github.com/jessicarick/resources/tree/master/tutorials). Note: this is kind of a hefty way of doing this, and you may be interested in just using Knitr to run R code in your LaTeX document instead for small tasks, as outlined in [this Overleaf tutorial](https://www.overleaf.com/learn/latex/Knitr).
2. **Using Make** - Not a tutorial of my own, but my colleague [Josh Harrison](https://github.com/jharrisonecoevo) has [developed a tutorial](https://github.com/JHarrisonEcoEvo/Reproducible_workflow_tutorial) on how to use [Make](https://www.gnu.org/software/make/manual/make.html) as a part of your reproducible research workflow. I'd highly recommend checking it out and contemplating how this tool might just make your life 100x easier!
3. **Running MSMC/PSMC Analyses** - I have put together a tutorial with scripts for running demographic analyses using PSMC/MSMC, which can be found in a [separate Github repository](https://github.com/jessicarick/msmc2_scripts).
4. **Using Snakemake** - Another tutorial that isn't my own, but from my colleague [Pavel Dimens](https://pdimens.github.io/), who has a very helpful introduction to Snakemake tutorial [on his website](https://pdimens.github.io/posts/snakemake101/), as well as fun Youtube videos ([A Gentle Introduction to Snakemake](https://www.youtube.com/playlist?list=PLm-MsgmJxrTTgTTvU_qrZlyvJQH1lmuI8)) walking through how to use this workflow manager to make your workflows more reproducible.

## Small scripts
1. **Calculating Reich FST** - I wrote a short R script for calculating the Reich-Patterson FST estimator from a genlight object, including bootstrapping confidence intervals, which can be found in a [separate Github repository](https://github.com/jessicarick/reich-fst).
2. **Removing individuals with missing data from Phylip alignments** - A short bash script that I wrote for removing individuals from Phylip alignments that have all missing data (or greater than some proportion that you specify), because I couldn't find a program or script ready-made that did this already. My use case is for removing individuals from gene tree alignments before inferring phylogenies in RAxML to input into ASTRAL. [remove_missing_phy.sh](https://github.com/jessicarick/resources/blob/master/tutorials/remove_missing_phy.sh)
